From bfc7ce3e3e808eeeddf955e8722c20e6120b861b Mon Sep 17 00:00:00 2001
From: Runxi Yu <me@runxiyu.org>
Date: Thu, 19 Feb 2026 14:15:54 +0800
Subject: [PATCH] os: use t.TempDir in root_test tempDirWithUnixSocket

tempDirWithUnixSocket used os.MkdirTemp("", ""), while other paths in
root consistency tests use t.TempDir().

If GOTMPDIR and the system temp dir are on different filesystems (such
as during abuild), TestRootConsistencyRename and TestRootConsistencyLink
may produce EXDEV.

Signed-off-by: Runxi Yu <me@runxiyu.org>
---
 src/os/root_test.go | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/src/os/root_test.go b/src/os/root_test.go
index f9fbd11575..c68d613dd9 100644
--- a/src/os/root_test.go
+++ b/src/os/root_test.go
@@ -1211,15 +1211,7 @@ var rootConsistencyTestCases = []rootConsistencyTest{{
 }}
 
 func tempDirWithUnixSocket(t *testing.T, name string) string {
-	dir, err := os.MkdirTemp("", "")
-	if err != nil {
-		t.Fatal(err)
-	}
-	t.Cleanup(func() {
-		if err := os.RemoveAll(dir); err != nil {
-			t.Error(err)
-		}
-	})
+	dir := t.TempDir()
 	addr, err := net.ResolveUnixAddr("unix", filepath.Join(dir, name))
 	if err != nil {
 		t.Skipf("net.ResolveUnixAddr: %v", err)
-- 
2.53.0

